datameta/validation.py

```diff
@@ -0,0 +1,204 @@
+# Copyright 2021 Universität Tübingen, DKFZ and EMBL for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from sqlalchemy.orm import joinedload
+from sqlalchemy import or_, and_
+from collections import defaultdict, Counter
+
+from . import resource, linting
+from .models import File, MetaDataSet, MetaDatumRecord, MetaDatum
+from .security import authz
+from .api.metadata import get_all_metadata
+from .utils import get_record_from_metadataset
+
+def validate_submission_access(db, db_files, db_msets, auth_user):
+    """Validates a submission with regard to
+
+    - Entities that cannot be found
+    - Entities the user does not have access to
+
+    Raises:
+        400 HTTPBadRequest
+    """
+    # Collect missing files
+    val_errors = [
+        ({ 'site' : file_id, 'uuid' : file_id }, None, "Not found")
+        for file_id, db_file in db_files.items()
+        if db_file is None
+    ]
+    # Collect authorization issues
+    val_errors += [
```

Comment:
`.extend()` or is the `+=` for aesthetics?

Reply:
I always concatenate lists with the `+` operator. Brief SO / google search even says it's faster for whatever reason. I also consider it more readable, but that's a matter of perspective I guess ;-)